pkg/provider/onboardbase/client/client.go

```diff
@@ -0,0 +1,402 @@
+
+	aesdecrypt "github.com/Onboardbase/go-cryptojs-aes-decrypt/decrypt"
+)
+
+type OnboardbaseClient struct {
+	baseURL             *url.URL
+	OnboardbaseAPIKey   string
+	VerifyTLS           bool
+	UserAgent           string
+	OnboardbasePassCode string
+	httpClient          *http.Client
+}
+
+type queryParams map[string]string
+
+type headers map[string]string
+type DeleteSecretsRequest struct {
+	SecretId string `json:"secretId,omitempty"`
+}
+
+type httpRequestBody []byte
+
+type Secrets map[string]string
+
+type RawSecret struct {
+	Key   string `json:"key,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+type RawSecrets []RawSecret
+
+type APIError struct {
+	Err     error
+	Message string
+	Data    string
+}
+
+type apiResponse struct {
+	HTTPResponse *http.Response
+	Body         []byte
+}
+
+type apiErrorResponse struct {
+	Messages []string
+	Success  bool
+}
+
+type SecretRequest struct {
+	Environment string
+	Project     string
+	Name        string
+}
+
+type SecretsRequest struct {
+	Environment string
+	Project     string
+}
+
+type secretResponseBodyObject struct {
+	Title string `json:"title,omitempty"`
+	Id    string `json:"id,omitempty"`
+}
+
+type secretResponseSecrets struct {
+	Id    string `json:"id"`
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type secretResponseBodyData struct {
+	Project     secretResponseBodyObject `json:"project,omitempty"`
+	Environment secretResponseBodyObject `json:"environment,omitempty"`
+	Team        secretResponseBodyObject `json:"team,omitempty"`
+	Secrets     []secretResponseSecrets  `json:"secrets,omitempty"`
+	Status      string                   `json:"status"`
+	Message     string                   `json:"string"`
+}
+
+type secretResponseBody struct {
+	Data    secretResponseBodyData `json:"data,omitempty"`
+	Message string                 `json:"message,omitempty"`
+	Status  string                 `json:"status,omitempty"`
+}
+
+type SecretResponse struct {
+	Name  string
+	Value string
+}
+
+type SecretsResponse struct {
+	Secrets Secrets
+	Body    []byte
+}
+
+func NewOnboardbaseClient(onboardbaseAPIKey, onboardbasePasscode string) (*OnboardbaseClient, error) {
+	tlsConfig := &tls.Config{
+		MinVersion: tls.VersionTLS12,
+	}
+	httpTransport := &http.Transport{
+		DisableKeepAlives: true,
+		TLSClientConfig:   tlsConfig,
+	}
+	client := &OnboardbaseClient{
+		OnboardbaseAPIKey:   onboardbaseAPIKey,
+		OnboardbasePassCode: onboardbasePasscode,
+		VerifyTLS:           true,
+		UserAgent:           "onboardbase-external-secrets",
+		httpClient: &http.Client{
+			Timeout:   10 * time.Second,
+			Transport: httpTransport,
+		},
+	}
+
+	if err := client.SetBaseURL("https://public.onboardbase.com/api/v1/"); err != nil {
+		return nil, &APIError{Err: err, Message: "setting base URL failed"}
+	}
+
+	return client, nil
+}
+
+func (c *OnboardbaseClient) BaseURL() *url.URL {
+	u := *c.baseURL
+	return &u
+}
+
+func (c *OnboardbaseClient) SetBaseURL(urlStr string) error {
```

Comment:
This function isn't called from anywhere else. So why check for HTTPS? It is called with a hardcoded API URL.

Are you planning on making it configurable?